universum/modules/output/html_output.py

```diff
@@ -1,106 +1,103 @@
 import os
 import re
 from datetime import datetime
+from re import Match
+from typing import Optional, List
 
 from ansi2html import Ansi2HTMLConverter
 
 from .base_output import BaseOutput
 
-
 __all__ = [
     "HtmlOutput"
 ]
 
 
 class HtmlOutput(BaseOutput):
-
     default_name = "universum_log.html"
 
-    def __init__(self, *args, log_name=default_name, **kwargs):
+    def __init__(self, *args, log_name: str = default_name, **kwargs):
         super().__init__(*args, **kwargs)
-        self._log_name = log_name
-        self._log_path = None
-        self.artifact_dir_ready = False
-        self._log_buffer = []
-        self._block_level = 0
-        self.module_dir = os.path.dirname(os.path.realpath(__file__))
-        self.ansi_converter = Ansi2HTMLConverter(inline=True, escaped=False)
-
-    def set_artifact_dir(self, artifact_dir):
-        self._log_path = os.path.join(artifact_dir, self._log_name)
-
-    def open_block(self, num_str, name):
-        opening_html = f'<input type="checkbox" id="{num_str}" class="hide"/>' + \
-            f'<label for="{num_str}"><span class="sectionLbl">'
-        closing_html = "</span></label><div>"
-        name_html = f'<span class="sectionTitle">{name}</span>'
-        self._log_line(f"{opening_html}{num_str} {name_html}{closing_html}", with_line_separator=False)
-        self._block_level += 1
+        self._log_name: str = log_name
+        self._log_path: Optional[str] = None
```

Comment:
not Optional[Path]?

Reply:
No. It is produced by os.path.join, which returns str.
Switching to Path or py is a different kind of refactoring.